### ISSUE TITLE
Question about the training speed acceleration.

### ISSUE THREAD

**Issue by ShunLu91** [KMnP/vpt#34]
Dear authors,

I sincerely thank you for your impressive work and invaluable contributions. I recently had the opportunity to read your paper, and I have a question regarding the practical implementation of VPT. Specifically, I am curious whether VPT has a training acceleration in practice.

As you know, VPT relies on a learnable shallow visual prompt, which necessitates a full gradient backward pass during training. However, this full backward pass can be more time-consuming than just fine-tuning the head layer. Consequently, I am curious to know whether VPT's training time is comparable to that of FULL-fine-tuning.

I would greatly appreciate it if you could kindly confirm whether my understanding is correct. Thank you again for your time and contribution to the field.

Best regards, 
Shun Lu

**Comment by Tsingularity** [comment]
Hi Shun,

Thanks for your interest in our work! And your understanding is correct. Although prompt tuning could save storage after training, it still requires similar amount of (even more when you have many prompt tokens) GPU memory compared to full-finetuning. You can find more discussion on this in our paper appendix "Empirical computational cost" part and figure 18.

**Comment by ShunLu91** [comment]
Thank you so much for taking the time to respond to my question. Your prompt and positive feedback means a lot to me!
